File: java/TaskieWithRetrofit/app/src/main/java/ada/osc/taskie/persistance/FakeDatabase.java
package ada.osc.taskie.persistance;

import java.util.ArrayList;
import java.util.List;

import ada.osc.taskie.model.Task;

public class FakeDatabase {

	private List<Task> mTasks;

	public FakeDatabase(){
		mTasks = new ArrayList<>();
	}

	public List<Task> getTasks(){
		return new ArrayList<>(mTasks);
	}

	public void save(Task task){
		mTasks.add(task);
	}

	public void save(List<Task> tasks){
		mTasks.addAll(tasks);
	}

	public void delete(Task task) {
		mTasks.remove(task);
	}
}

File: kotlin/Taskie/app/src/main/java/ada/osc/taskie/persistance/FakeDatabase.kt
package ada.osc.taskie.persistance

import ada.osc.taskie.model.Task

class FakeDatabase {

    val tasks = mutableListOf<Task>()

    fun save(task: Task) = tasks.add(task)
    fun save(tasks: List<Task>) = this.tasks.addAll(tasks)
    fun delete(task: Task) = tasks.remove(task)
}
